src/azul/service/responseobjects/hca_response_v5.py

```diff
@@ -292,15 +292,19 @@ def _write_tsv(self, output: IO[str]) -> None:
                 writer.writerow(row)
 
     def _create_bdbag_archive(self) -> str:
-        with TemporaryDirectory() as bag_path:
+        with TemporaryDirectory() as temp_path:
+            bag_path = os.path.join(temp_path, 'manifest')
+            os.makedirs(bag_path)
             bdbag_api.make_bag(bag_path)
             with open(os.path.join(bag_path, 'data', 'samples.tsv'), 'w') as samples_tsv:
                 self._write_bdbag_samples_tsv(samples_tsv)
             bag = bdbag_api.make_bag(bag_path, update=True)  # update TSV checksums
             assert bdbag_api.is_bag(bag_path)
             bdbag_api.validate_bag(bag_path)
             assert bdbag_api.check_payload_consistency(bag)
-            return bdbag_api.archive_bag(bag_path, 'zip')
+            _, temp_path = mkstemp()
```

Comment:
You are leaking a file handle.

```suggestion
            temp, temp_path = mkstemp()
            temp.close()
```

Reply:
I don't think that file handle is leaking. mkstemp() returns an (integer, str).